### PR TITLE
Implement Read method for BlockSource

### LINKED ISSUE #194: Blocks Support: Create prefect_block resource - Read()

### PR DESCRIPTION
**Pull request by mitchnielsen**
Implements BlockSource.Read method.

Closes https://github.com/PrefectHQ/terraform-provider-prefect/issues/194

### CODE REVIEW

**Comment by parkedwards** [comment]
@mitchnielsen i updated the linked issue in your PR description

File: internal/provider/resources/block.go [hunk 1]
```diff
@@ -127,9 +129,64 @@ func (r *BlockResource) Create(ctx context.Context, req resource.CreateRequest,
...
+
+		return
+	}
+
+	state.ID = types.StringValue(block.ID.String())
+	state.Created = customtypes.NewTimestampPointerValue(block.Created)
+	state.Updated = customtypes.NewTimestampPointerValue(block.Updated)
+	state.Name = types.StringValue(*block.Name)
+	state.Type = types.StringValue(block.BlockType.Slug)
+	state.Data = jsontypes.NewNormalizedValue(block.Data)
```
Comment (mitchnielsen) [inline internal/provider/resources/block.go:183]: We end up using this in other methods as well, like `Update`, so in a future PR I'll move this to a reusable function.

File: internal/provider/resources/block.go [hunk 2]
```diff
@@ -127,9 +129,64 @@ func (r *BlockResource) Create(ctx context.Context, req resource.CreateRequest,
 }
 
 // Read refreshes the Terraform state with the latest data.
-//
-//nolint:revive // TODO: remove this comment when method is implemented
 func (r *BlockResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state BlockResourceModel
```
Comment (mitchnielsen) [inline internal/provider/resources/block.go:133]: Sometimes we call this `state`, sometimes `model`. The Terraform tutorial calls it `state`, which felt a little more intuitive for me when reading the rest of the function. But I'm good either way.

File: internal/provider/resources/block.go [hunk 3]
```diff
@@ -127,9 +129,64 @@ func (r *BlockResource) Create(ctx context.Context, req resource.CreateRequest,
...
+	var blockID uuid.UUID
+	blockID, err = uuid.Parse(state.ID.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddAttributeError(
+			path.Root("id"),
+			"Error parsing Block ID",
+			fmt.Sprintf("Could not parse block ID to UUID, unexpected error: %s", err.Error()),
+		)
+
+		return
```
Comment (parkedwards) [inline internal/provider/resources/block.go:165]: it's worth mentioning here before we go too far down the rabbit hole - just like you have it here, anything that the user interfaces with here will be the `block`, whereas the API resources that constitute it are the `block_document` / `block_type` / `block_schema` etc.

**Review by parkedwards — APPROVED** [review]
nice!

File: internal/provider/resources/block.go (hunk 3 above)
Comment (mitchnielsen) [reply internal/provider/resources/block.go:165]: Good callout 👍🏼

File: internal/provider/resources/block.go (hunk 2 above)
Comment (parkedwards) [reply internal/provider/resources/block.go:133]: yeah, i think we should eventually do away with the `model` naming.  IMO we should only have `state` or `plan` variables, based on which framework function (eg. `req.State.Get()` or `req.Plan.Get()`) is doing the unmarshalling

Comment (mitchnielsen) [reply internal/provider/resources/block.go:133]: Agreed 👍🏼 Opened https://github.com/PrefectHQ/terraform-provider-prefect/issues/195 so we can come back to that at some point.

**Pull request merged**